### ISSUE TITLE
Disable strict SSL in ./node_modules/vscode/bin/install script?

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode-extension-vscode#57]
_From @emackey on March 3, 2017 19:16_

I'm on a corporate network (not a proxy) that imposes its own self-signed SSL certificate on outbound https traffic.

I already have this in my `settings.json`, which is insecure but at least allows me to install vscode extensions:

```
    "http.proxyStrictSSL": false
```

Likewise, I can turn off strict ssl to install npm modules, with `npm set strict-ssl false`.  This is git bash on Windows:

```
$ npm config ls -l | grep strict-ssl
strict-ssl = false
```

```
$ cat package.json
{
  "name": "test-vscode-install",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "engines": {
    "vscode": "^1.0.0"
  },
  "scripts": {
        "vscode:prepublish": "node ./node_modules/vscode/bin/compile",
        "compile": "node ./node_modules/vscode/bin/compile -watch -p ./",
        "postinstall": "node ./node_modules/vscode/bin/install"
  },
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "typescript": "^2.2.1",
    "vscode": "^1.0.5"
  }
}
```

```
$ npm install

> test-vscode-install@1.0.0 postinstall C:\path\test-vscode-install
> node ./node_modules/vscode/bin/install

Detected VS Code engine version: ^1.0.0
Error installing vscode.d.ts: Error: self signed certificate in certificate chain
```

So, to summarize:

* :+1:  vscode settings.json is configured for insecure SSL, and can install extensions from the marketplace.
* :+1: npm is configured for insecure SSL, and can install ordinary npm packages without trouble.
* :x: The vscode "postinstall" node script is still choking on the self-signed cert.

Is there a way to fix this?  Thanks!

_Copied from original issue: Microsoft/vscode#21926_

**Comment by bpasero** [comment]
_From @joaomoreno on March 6, 2017 10:46_

@bpasero The code to fetch `vscode.d.ts` should maybe pay attention to npm's value of `strict-ssl`?

**Comment by bpasero** [comment]
@emackey to verify the fix would work can you open the file `node_modules/vscode/lib/shared.js` in your extension folder and change the options bag to this:

```json
var options = {
    url: url,
    strictSSL: false,
    headers: headers
};
```

And then run npm install again to see if this time it fetches the vscode.d.ts properly?

**Comment by emackey** [comment]
@bpasero I can confirm that the options change fixes the problem for me.  Thanks for taking this on!

**Comment by bpasero** [comment]
Fixed via 1.0.6

**Comment by camwar11** [comment]
@bpasero it seems that there is a bug in npm when setting environment variables based on config variables to false (see [here](https://github.com/electron-userland/electron-prebuilt/pull/97)). I think this issue still exists due to this fact (and that I am still running into it)

**Comment by bpasero** [comment]
@camwar11 good catch, want to jump in with a PR?

**Comment by camwar11** [comment]
@bpasero Sure, I should be able to work it tonight.
